packages/fuel-gauge/src/transaction-response.test.ts

```diff
@@ -75,25 +83,33 @@ describe('TransactionSummary', () => {
   });
 
   it('should ensure waitForResult always waits for the transaction to be processed', async () => {
-    const destination = Wallet.generate({
-      provider,
+    const { cleanup, ip, port } = await launchNode({
+      args: ['--poa-interval-period', '10s'],
```

Comment:
How many transactions are being executed in this node? 

If takes 10 seconds to process a transaction and supposedly the test executes 3 transactions, this means that this test will take at least 30 seconds to finish.

Reply:
I changed it to 5s. I'd still like to have it big enough to actually test the subscription behavior properly.

Reply:
Would it be possible to spy on the surroundings of the execution and ensure functionality without using `--poa`?

Reply:
This is the only way to be 100% certain that a transaction who's processing is taking longer on the _node_ will be properly awaited. IMO this needs to be an integration test, nothing less.

Reply:
There is no way we can accommodate a test with a 25s timeout. 😅 

Let's try thinking of other ways. We need it to be faster, without using `--poa`.

Requests to the node are always async, delays shouldn't be necessary to validate the behavior.

Reply:
I lowered to `--poa-interval-period 3s` and the test takes 11s to complete on my machine. I removed the 25s timeout. That's acceptable in my books, especially for such a critical feature.
This is the _only_ test in our test suite that's **fully** verifying that our integration with the fuel node's server-sent events functionality works in cases where the transaction takes longer to process and its status isn't a success or failure right away. This is a real-world scenario that is to be expected; a node might have multiple transactions on it at one moment and delay the execution some.

To test this without the node, we'd have to mock `fetch` to return a `Response` object who's body would be a [`ReadableStream`-like object implementing a custom async iterator](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream#async_iteration) that would:
1. Return the appropriate JSON with `TransactionStatus.SubmittedStatus`,
2. Sleep for a bit?
3. Return the appropriate JSON with `TransactionStatus.SuccessStatus`. 

This would probably work, but it wouldn't protect us from false positives due to possible breaking changes from one `fuel-core` version to another. IMO the benefits of a full integration test in this case outweigh the cost of a relatively slow test, especially because it's the only test of this kind in the code base. 

With all that said - if you still feel strongly about it, I'll implement the mocking.

Reply:
Much better!